src/bluehawk/actions/run.ts

```diff
@@ -0,0 +1,248 @@
+  configPath?: string;
+}
+
+export interface Config {
+  commands: ConfigAction[];
+}
+
+export interface ConfigAction {
+  command: "snip" | "copy" | "check";
+  source: string;
+  destination: string;
+  ignore?: string[];
+  state?: string;
+  format?: "rst" | "md" | "docusaurus";
+  json?: boolean;
+}
+
+export const run = async (
+  args: WithActionReporter<ConfigArgs>
+): Promise<void> => {
+  const { configPath, reporter } = args;
+  const currentWorkingDirectory = Path.resolve(process.cwd());
+  const configFileName = "bluehawk.config.yaml";
+
+  let rootConfig: Config | undefined = undefined;
+  let rootConfigPath = configPath ? configPath : undefined;
+
+  // Get and parse root config file
+  const parseConfigFile = async (configFilePath: string) => {
+    const configFile = await fsp.readFile(configFilePath, "utf-8");
+    const parsedConfigFile: Config = YAML.parse(configFile);
+
+    return parsedConfigFile;
+  };
+
+  // Look for "bluehawk.config.yaml" in current working directory. If not
+  // there, traverse up the file path to system root.
+  const getRootConfigFile = async (
+    directory = currentWorkingDirectory
+  ): Promise<Config> => {
+    const ls = await fsp.readdir(directory);
+
+    if (ls.includes(configFileName)) {
+      const configFilePath = Path.join(directory, configFileName);
+      rootConfigPath = configFilePath;
+
+      return parseConfigFile(configFilePath);
+    } else if (directory == "/") {
+      throw new Error(`Couldn't find ${configFileName}`);
+    } else {
+      return getRootConfigFile(Path.resolve(directory, ".."));
+    }
+  };
+
+  if (configPath) {
+    rootConfig = await parseConfigFile(configPath[0]);
+  } else {
+    rootConfig = await getRootConfigFile();
+  }
+
+  // Process root config file
+  if (rootConfig) {
+    reporter.onFileParsed({
+      inputPath: rootConfigPath!,
+      isConfig: true,
+    });
+
+    // Run config actions
+    for (let index = 0; index < rootConfig.commands.length; index++) {
+      const { command, source, destination, ignore, state, format, json } =
+        rootConfig.commands[index];
+
+      reporter.onActionProcessed({
+        inputPath: rootConfigPath!,
+        name: command,
+      });
+
+      switch (command) {
+        case "snip":
+          await snip({
+            paths: [source],
+            output: destination,
+            ignore: ignore,
+            state: state,
+            format: format,
+            reporter,
+            waitForListeners: true,
+          });
+
+          break;
+
+        case "check":
+          await check({
+            paths: [source],
+            ignore: ignore,
+            json: json,
+            reporter,
+          });
+
+          break;
+
+        case "copy":
+          await copy({
+            reporter,
+            output: destination,
+            rootPath: source,
+            waitForListeners: true,
+          });
+
+          break;
+
+        default:
```

Comment:
This isn't as allowing for extensibility as I would hope - I think a better approach would be to collect all the possible actions somewhere, allow plugins to add to the list of available actions, then use the action list here.

I think such a thing can be done in a subsequent PR.